core/drivers/stm32_bsec.c

```diff
@@ -153,7 +153,8 @@ static uint32_t otp_bank_offset(uint32_t otp_id)
 
 static vaddr_t bsec_base(void)
 {
-	return io_pa_or_va_secure(&bsec_dev.base);
+	return io_pa_or_va_secure(&bsec_dev.base, BSEC_IP_MAGIC_ID_OFF +
+				  sizeof(uint32_t));
```

Comment:
prefer `return io_pa_or_va_secure(&bsec_dev.base, 1);` here.
BSEC iomem does not cross 4KB boundaries.